presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -269,7 +270,9 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                 continue;
             }
             ConnectorId connectorId = transactionManager.getOptionalCatalogMetadata(transactionId, catalogName)
-                    .orElseThrow(() -> new PrestoException(NOT_FOUND, "Catalog does not exist: " + catalogName))
+                    .orElseThrow(() ->
+                            new PrestoException(NOT_FOUND,
+                                    "Catalog " + catalogName + " does not exist for session property: " + getFirst(catalogProperties.keySet(), null)))
```

Comment:
It's better to use `iterator().next()` when the item is required. `getFirst()` will return a default which isn't what we want here.